Makefile

```diff
@@ -1,33 +1,57 @@
+IMAGE_NAME ?= mlops-app-crunchy-postgres
+IMAGE_TAG ?= latest
+
 .PHONY: all clean test lint format
 all clean test lint format:
 
 SHELL := /bin/sh -e
 
-venv:
-	python -m venv venv
-	. venv/bin/activate; \
-	python -m pip install --upgrade pip
-
-.PHONY: setup
-setup: venv
-	. venv/bin/activate; \
-	pip install -r requirements/base.txt
+.PHONY: install setup
+install setup: poetry.lock
+	poetry config virtualenvs.in-project true
+	poetry install --with dev
+	poetry run pre-commit install;
 
 .PHONY: install
 install:
-	pip install pre-commit;
-	pre-commit install;
+	poetry config virtualenvs.in-project true
+	poetry install --with dev
+	poetry run pre-commit install;
 
-.PHONY: lint format
-lint format:
+.PHONY: format
+format:
 ifdef CI
-	pre-commit run --all-files --show-diff-on-failure
+	poetry run pre-commit run --all-files --show-diff-on-failure
 else
-	pre-commit run --all-files || pre-commit run --all-files
+	# automatically fix the formatting issues and rerun again
+	poetry run pre-commit run --all-files || poetry run pre-commit run --all-files
 endif
 
-.PHONY: test
-test:
+.PHONY: lint
+lint: format
+	poetry run mypy .apolo
+
+.PHONY: test-unit
+test-unit:
+	poetry run pytest -vvs --cov=.apolo --cov-report xml:.coverage.unit.xml .apolo/tests/unit
+
+.PHONY: test-integration
+test-integration:
+	poetry run pytest -vv --cov=.apolo --cov-report xml:.coverage.integration.xml .apolo/tests/integration
+
+.PHONY: build-hook-image
+build-hook-image:
+	docker build \
+		-t $(IMAGE_NAME):latest \
+		-f hooks.Dockerfile \
+		.;
+
+.PHONY: push-hook-image
+push-hook-image:
+	docker tag $(IMAGE_NAME):latest ghcr.io/neuro-inc/$(IMAGE_NAME):$(IMAGE_TAG)
+	docker push ghcr.io/neuro-inc/$(IMAGE_NAME):$(IMAGE_TAG)
 
-.PHONY: clean
-clean:
+.PHONY: gen-types-schemas
+gen-types-schemas:
+	app-types dump-types-schema .apolo/src/apolo_apps_postgresql postgresql PostgresInputs .apolo/src/apolo_apps_postgresql/schemas/PostgresInputs.json
```

Comment:
Can we delete `app_type` from here? AFAIK, it is used only for logging

Reply:
I think yes, let's clear that up